[SPARK-51839] Support `except(All)?/intersect(All)?/union(All)?/unionByName` in `DataFrame`

dongjoon-hyun opened the pull request:
### What changes were proposed in this pull request?

This PR aims to support seven `Set`-related `DataFrame` APIs.
- `except`
- `exceptAll`
- `intersect`
- `intersectAll`
- `union`
- `unionAll`
- `unionByName`

### Why are the changes needed?

For feature parity.

### Does this PR introduce _any_ user-facing change?

No, this is a new addition to the unreleased version.

### How was this patch tested?

Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

viirya commented on Sources/SparkConnect/DataFrame.swift:
```diff
@@ -499,6 +499,86 @@ public actor DataFrame: Sendable {
+      right,
+      SetOpType.union,
+      byName: true,
+      allowMissingColumns: allowMissingColumns
```
Do we need to set `isAll: true` for `unionByName`?

dongjoon-hyun commented:
Thank you! You are right. I addressed your comment by fixing it and adding a test coverage, @viirya .
> Do we need to set isAll: true for unionByName like union?

dongjoon-hyun commented:
Merged to main~
